serializer_impl.hh

```diff
@@ -851,6 +860,23 @@ public:
         }
     }
 
+    vector_deserializer(vector_deserializer&& o) = default;
+    vector_deserializer& operator=(const vector_deserializer& o) = delete;
+    vector_deserializer& operator=(vector_deserializer&& o) = default;
+    vector_deserializer(const vector_deserializer& o) = delete;
+
+    template<bool B> vector_deserializer<T, B> to_forward() &&;
+    template<>
+    vector_deserializer<T, true> to_forward<true>() && {
+        static_assert(IsForward, "cannot convert reverse vector deserializer into forward one");
```

Comment:
Note: nicer to have `requires IsForward` as a static_assert implies something wrong with the callee, not the caller.

Reply:
Also: does it have to be a template? I guess it does with static_assert to prevent early assertion.

Reply:
I'm sure it can be improved. But first we need to agree if we want the thing or not.